libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -447,8 +448,24 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 		m_evm15,
 		m_identifierAccess,
 		m_useNamedLabelsForFunctions,
-		m_context
+		m_context,
+		_function.returnVariables,
+		m_assembly.newLabelId()
 	);
+	subTransform.m_scope = virtualFunctionScope;
+
+	if (m_allowStackOpt)
+		// Immediately delete entirely unused parameters.
+		for (auto const& v: _function.parameters | boost::adaptors::reversed)
+		{
+			auto& var = std::get<Scope::Variable>(virtualFunctionScope->identifiers.at(v.name));
+			if (util::valueOrDefault(m_context->variableReferences, &var, 0u) == 0)
+				subTransform.deleteVariable(var);
+		}
+
+	if (!m_allowStackOpt || _function.returnVariables.empty())
```

Comment:
`|| .empty()` sounds like an unnecessary special case. This is also for initializing the return stack height, right?

Reply:
Yes, if I don't have return variables, then initializing the return stack height is never triggered later otherwise, so having the special case here for that seemed easiest.